### PR TITLE
Improved test coverage for preemption in pkg/detector

### PR DESCRIPTION
**Pull request by NishantBansal2003**
**What type of PR is this?**
/kind feature
/kind failing-test

**What this PR does / why we need it**:
The test case coverage for pkg/detector/preemption.go has been improved.
**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
To verify the changes in the pkg/detector/ directory run the following commands:
```
go test ./... -coverprofile=coverage.out
go tool cover -html=coverage.out -o coverage.html
open coverage.html
```
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by NishantBansal2003** [comment]
PTAL
cc: @XiShanYongYe-Chang

**Comment by XiShanYongYe-Chang** [comment]
Thanks
/assign

**Comment by yashpandey06** [comment]
@NishantBansal2003 please correct the fixes here too .
Thanks

**Comment by NishantBansal2003** [comment]
Could you please review this? Thank you!
cc: @XiShanYongYe-Chang

File: pkg/detector/preemption_test.go [hunk 1]
```diff
@@ -0,0 +1,659 @@
...
+		t.Run(tt.name, func(t *testing.T) {
+			fakeClient := tt.setupClient().Build()
+			stopCh := make(chan struct{})
+			defer close(stopCh)
+			var fakeDynamicClient *dynamicfake.FakeDynamicClient
+			if tt.objects != nil {
+				fakeDynamicClient = dynamicfake.NewSimpleDynamicClient(scheme, tt.objects...)
+			} else {
+				fakeDynamicClient = dynamicfake.NewSimpleDynamicClient(scheme)
+			}
```
Comment (XiShanYongYe-Chang) [inline pkg/detector/preemption_test.go:346]: It's okay to just write that: `fakeDynamicClient = dynamicfake.NewSimpleDynamicClient(scheme, tt.objects...)`

File: pkg/detector/preemption_test.go [hunk 2]
```diff
@@ -0,0 +1,659 @@
...
```
Comment (XiShanYongYe-Chang) [inline pkg/detector/preemption_test.go:639]: ditto

**Review by XiShanYongYe-Chang — COMMENTED** [review]
Good job.

**Comment by NishantBansal2003** [comment]
Made the required changes, PTAL
cc: @XiShanYongYe-Chang

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**